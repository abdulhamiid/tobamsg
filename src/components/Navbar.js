import React from 'react';
import {
  Link,
} from 'react-router-dom';

function Navbar() {
  return (
    <nav>
      <div>
        <header>
          <h1><Link to="/">Foodieland</Link></h1>
        </header>
        <ul className="navbar">
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/recipes">Recipes</Link>
          </li>
          <li>
            <Link to="/blog">Blog</Link>
          </li>
          <li>
            <Link to="/contact">Contact</Link>
          </li>
          <li>
            <Link to="/about">About us</Link>
          </li>
        </ul>
        <ul>
            <li>
                <a>
                    <img src="001-facebook.svg" />
                </a>
            </li>
            <li>
                <a>
                    <img src="003-twitter.svg" />
                </a>
            </li>
            <li>
                <a>
                    <img src="004-instagram.svg" />
                </a>
            </li>
        </ul>
      </div>
    </nav>
  );
}

export default Navbar;