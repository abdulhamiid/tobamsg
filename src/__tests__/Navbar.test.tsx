import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar';

// TypeScript ensures `tree` is of type `ReactTestRendererJSON | ReactTestRendererJSON[] | null`
it('Test Navigation renders correctly', () => {
  const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer
    .create(
      <Router>
        <Navbar />
      </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
