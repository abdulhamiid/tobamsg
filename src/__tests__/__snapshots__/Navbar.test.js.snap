// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navigation renders correctly 1`] = `
<nav>
  <div
    id="navbar"
  >
    <header>
      <h1
        id="header"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Foodieland
          <span
            className="text-[#ff6363]"
          >
            .
          </span>
        </a>
      </h1>
    </header>
    <ul
      className="nav-links"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/recipes"
          onClick={[Function]}
        >
          Recipes
        </a>
      </li>
      <li>
        <a
          href="/blog"
          onClick={[Function]}
        >
          Blog
        </a>
      </li>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About us
        </a>
      </li>
    </ul>
    <ul
      className="nav-links"
    >
      <li>
        <a
          href=""
        >
          <img
            src="001-facebook.svg"
          />
        </a>
      </li>
      <li>
        <a
          href=""
        >
          <img
            src="003-twitter.svg"
          />
        </a>
      </li>
      <li>
        <a
          href=""
        >
          <img
            src="instagram.svg"
          />
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
